launch/org.eclipse.cdt.docker.launcher/META-INF/MANIFEST.MF

```diff
@@ -29,6 +30,7 @@ Require-Bundle: org.eclipse.ui,
  org.eclipse.core.databinding;bundle-version="1.6.0",
  org.eclipse.core.databinding.property;bundle-version="1.6.0",
  org.eclipse.launchbar.ui;bundle-version="2.2.0",
+ org.mandas.docker-client;bundle-version="3.2.1",
```

Comment:
I wasn't able to compile without it.

Reply:
Maybe you could post the compilation error.  It could be resolution of a var that you use.  In any case, you shouldn't require it and if it is absolutely required, then the code belongs in Linux Tools.

Reply:
Ok, I did some experiments and I'm finally hopelessly confused.

https://github.com/cmorty/cdt/blob/903178412221e59d7961f639761cb5ff31656ceb/launch/org.eclipse.cdt.docker.launcher/src/org/eclipse/cdt/internal/docker/launcher/ContainerLaunchUtils.java#L125

Eclipse complains:
>   The type org.mandas.docker.client.exceptions.DockerCertificateException cannot be resolved. It is indirectly referenced from required .class files

So if I replace
```java
    connection.pullImage(imageName, prghandler);
```
with
```java
    ((IDockerConnection) connection).pullImage(imageName, prghandler);
```
the error goes away. :(

Reply:
I don't understand why sometimes JDT (javac too?) gives the indirectly referenced error like that. But in this case the fix does indeed belong in LinuxTools to resolve the problem "properly". pullImage is throwing org.mandas.docker.client.exceptions.DockerCertificateException which means the org.mandas API is leaking out into Linuxtools API.

Reply:
My guess is that by changing it to type IDockerConnection it works because it is IDockerConnection that has the import so some resolution of class changes because of that.

Reply:
Unfortunately removing that throws (perhaps by wrapping it in a Linux Tools specific exception) is an API breaking change as [adding or removing](https://wiki.eclipse.org/Evolving_Java-based_APIs_2) a checked exception from the throws declaration breaks compatibility.

@jjohnstn Which alternative is best for you in this case? Allow the require in CDT for org.mandas, or make the change in LinuxTools now? A quick check of all the interfaces in org.eclipse.linuxtools.docker.core indicates this is the only one that seems to leak out API

Reply:
Or option 3 of just using the casting workaround.

Reply:
@cmorty Ok, there's something up with PDE.  It seems to be confused by an overloaded method.  In your case, you are using a method pullImage() that takes two parameters and returns public exceptions.  There is another pullImage() which takes 3 parameters and it throws DockerCertificateException which is part of org.mandas.docker-client.  The one method does not call the other.  With your call, Eclipse thinks you have a reference to DockerCertificateException that requires the docker-client reference.  I did an experiment where I created another method: pullImageWithHandler() method which has the same signature and logic as the pullImage() method you are using.  Works fine and the bundle is not required.  I am ok with checking this in if Eclipse is correct in checking the overloaded methods or there isn't a simple way to stop the error.  I also have a version of your code that calls the new method and returns IStatus so I can do that change too.

Reply:
Sorry @jonahgraham, didn't see your comments.  As mentioned, it seems that there is some issue with overloaded methods.  @cmorty is using a valid call that doesn't throw the DockerCertificateException but Eclipse seems to confuse that with another pullImage() method that does reference it.  If I add a new method with a different name, it works fine without requiring org.mandas....  I would actually like to deprecate the other method as it should have wrapped the mandas exception being a public interface.

Reply:
Ok. I'll do the cast and write a comment. That seems like the simplest solution to me.